### PR TITLE
Implement subscription validation disabling for OAuth tokens

### PR DESCRIPTION
**Pull request by RakhithaRR**
## Purpose

Related issue: https://github.com/wso2/api-manager/issues/3011

This PR adds the UI implementation to support subscription validation disabling for Oauth tokens.

## Implementation

- Adds a new internal REST API resource to create an internal subscription when subscription validation is disabled for an API.
-  Introduce new internal business plans to use when subscription validation is disabled.
-  Create a new internal subscription on the first API invocation if an internal subscription does not exist.
- Allow valid external tokens to bypass subscription validation process when validation is disabled.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.internal.service/src/main/java/org/wso2/carbon/apimgt/internal/service/impl/SubscribeInternalApiServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
+package org.wso2.carbon.apimgt.internal.service.impl;
```
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.internal.service/src/main/java/org/wso2/carbon/apimgt/internal/service/impl/SubscribeInternalApiServiceImpl.java:1]: License header

File: features/apimgt/org.wso2.carbon.apimgt.core.feature/src/main/resources/conf_templates/templates/repository/conf/api-manager.xml.j2 [hunk 2]
```diff
@@ -1106,6 +1106,7 @@
                 <Silver>2000</Silver>
                 <Bronze>1000</Bronze>
                 <Unauthenticated>60</Unauthenticated>
+                <Default_Subscriptionless>100000</Default_Subscriptionless>
```
Comment (chamilaadhi) [inline features/apimgt/org.wso2.carbon.apimgt.core.feature/src/main/resources/conf_templates/templates/repository/conf/api-manager.xml.j2:1109]: Shall we change this to the naming format used in other elements
```suggestion
                <DefaultSubscriptionless>100000</DefaultSubscriptionless>
```

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/SubscriptionValidationDAO.java [hunk 3]
```diff
@@ -1565,4 +1569,88 @@ public List<API> getAllApisByLabel(String gatewayLabel, Boolean expand) {
...
+                preparedStForInsert.setString(5, APIConstants.SubscriptionCreatedStatus.SUBSCRIBE);
+                preparedStForInsert.setString(6, subscriber);
+
+                Timestamp timestamp = new Timestamp(System.currentTimeMillis());
+                preparedStForInsert.setTimestamp(7, timestamp);
+                preparedStForInsert.setTimestamp(8, timestamp);
+                preparedStForInsert.setString(9, subscriptionUUID);
+
+                preparedStForInsert.executeUpdate();
+                try (ResultSet rs = preparedStForInsert.getGeneratedKeys()) {
```
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/SubscriptionValidationDAO.java:1642]: use 'connection.setAutoCommit(false);' and  'connection.commit();'

File: components/apimgt/org.wso2.carbon.apimgt.internal.service/src/main/java/org/wso2/carbon/apimgt/internal/service/impl/SubscribeInternalApiServiceImpl.java (hunk 1 above)
Comment (RakhithaRR) [reply components/apimgt/org.wso2.carbon.apimgt.internal.service/src/main/java/org/wso2/carbon/apimgt/internal/service/impl/SubscribeInternalApiServiceImpl.java:19]: Fixed with https://github.com/wso2/carbon-apimgt/pull/12539/commits/9cd326adf51117359068bd6457efdaf33d209aea

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/SubscriptionValidationDAO.java (hunk 3 above)
Comment (RakhithaRR) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/SubscriptionValidationDAO.java:1643]: Fixed with https://github.com/wso2/carbon-apimgt/pull/12539/commits/9cd326adf51117359068bd6457efdaf33d209aea

File: features/apimgt/org.wso2.carbon.apimgt.core.feature/src/main/resources/conf_templates/templates/repository/conf/api-manager.xml.j2 (hunk 2 above)
Comment (RakhithaRR) [reply features/apimgt/org.wso2.carbon.apimgt.core.feature/src/main/resources/conf_templates/templates/repository/conf/api-manager.xml.j2]: Fixed with https://github.com/wso2/carbon-apimgt/pull/12539/commits/9cd326adf51117359068bd6457efdaf33d209aea

**Comment by RakhithaRR** [comment]
The failing tests are fixed through https://github.com/wso2/product-apim/pull/13527

**Review by tharindu1st — APPROVED** [review]

**Pull request merged**